lua/telescope/_extensions/projects.lua

```diff
@@ -165,8 +165,17 @@ local function projects(opts)
       map("i", "<c-w>", change_working_directory)
 
       local on_project_selected = function()
+        if(config.options.before_project_selection_callback) then
+          config.options.before_project_selection_callback()
+        end
```

Comment:
I believe this function should return a boolean, and if the callback returns `false`, then it should fall through to the default behavior.

This will allow you to do something custom when a project is selected without preventing the default behavior.

It brings up some interesting thoughts about this, though. The default behavior is simply to change the directory using the `scope_chdir` scope and optionally printing if `silent_chdir` is `false`.

In my use-case, I use the selected project to set the tab page name, but I still want it to present the files just as it did before

Reply:
AFAIK this doesn't affect the default behavior at all. It just executes a function before the default behavior occurs if the user has defined something they want to execute before selection occurs. Any manipulation of what might happen to the default selection behavior would be the user's responsibility to handle & customize (as it should be).

Please let me know if I've misunderstood your message.

Additionally, feel free to use this as a starting point for any further customizations you'd like to make but, given this project doesn't seem to be active anymore, I don't think this is ever getting merged.

Reply:
oh you are right. lgtm.

Yeah seems like a dead project. Like everyone else I forked it. Also changed the name to `noun` just to give it a name that works better with modern package managers (mainly, one that matches the repo name so the setup function can be called automatically). Been going through and adding some of the PRs here.

https://github.com/synic/noun.nvim